addons/survey/models/survey_user.py

```diff
@@ -250,92 +248,94 @@ def write(self, vals):
             vals.update({'answer_score': self.env['survey.question.answer'].browse(int(suggested_answer_id)).answer_score})
         return super(SurveyUserInputLine, self).write(vals)
 
-    def _get_save_line_values(self, answer, answer_type):
-        if not answer or (isinstance(answer, str) and not answer.strip()):
-            return {'answer_type': None, 'skipped': True}
-        if answer_type == 'suggestion':
-            return {'answer_type': answer_type, 'suggested_answer_id': answer}
-        value = float(answer) if answer_type == 'numerical_box' else answer
-        return {'answer_type': answer_type, 'value_' + answer_type: value}
-
     @api.model
-    def save_lines(self, user_input_id, question, answer, comment=None):
+    def save_lines(self, user_input, question, answer, comment=None):
```

Comment:
Starting to wander if this should be on the survey.user_input model itself.

That way we would not need to move the "user_input" along everywhere when calling sub-methods.

Reply:
I agree and wanted to do it but did not do it to avoid too much diff. As we have 2K lines diff let us make it still bigger.